From d8a783aa9b945aaef55c872adb5f3a63a7fb815c Mon Sep 17 00:00:00 2001
From: Xueqiang Gong <gongx@qti.qualcomm.com>
Date: Thu, 11 Sep 2025 14:57:12 +0800
Subject: [PATCH] msi: revert code genirq/msi: cache the latest msi msg

Upstream-Status: Pending
Signed-off-by: Xueqiang Gong <gongx@qti.qualcomm.com>
---
 kernel/irq/msi.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/irq/msi.c b/kernel/irq/msi.c
index a83fdc053c2b..38f13150a563 100644
--- a/kernel/irq/msi.c
+++ b/kernel/irq/msi.c
@@ -633,9 +633,9 @@ static unsigned int msi_domain_get_hwsize(struct device *dev, unsigned int domid
 static inline void irq_chip_write_msi_msg(struct irq_data *data,
 					  struct msi_msg *msg)
 {
-	struct msi_desc *desc = irq_data_get_msi_desc(data);
+	//struct msi_desc *desc = irq_data_get_msi_desc(data);
 
-	desc->msg = *msg;
+	//desc->msg = *msg;
 	data->chip->irq_write_msi_msg(data, msg);
 }
 
-- 
2.34.1

